test/test_api_responses.py

```diff
@@ -0,0 +1,71 @@
+    'VulnerabilitiesApi:cve_reports_uid_get': ('VulnerabilitiesApi', 'cve_reports_uid_get', {'uid': '6b95e65d42a546629cb0a23bc022ecde'}, 'FullCveSchema', 'https://api.intel471.com/v1/cve/reports/6b95e65d42a546629cb0a23bc022ecde'),
+    'WatchersApi:watcher_groups_get': ('WatchersApi', 'watcher_groups_get', {}, 'WatcherGroupResponse', 'https://api.intel471.com/v1/watcherGroups'),
+    'WatchersApi:watcher_groups_group_uid_get': ('WatchersApi', 'watcher_groups_group_uid_get', {'group_uid': '3d0b4432-7390-4ae5-92be-52c3e39dc775'}, 'SimpleWatcherGroupSchema', 'https://api.intel471.com/v1/watcherGroups/3d0b4432-7390-4ae5-92be-52c3e39dc775'),
+    'WatchersApi:watcher_groups_group_uid_watchers_get': ('WatchersApi', 'watcher_groups_group_uid_watchers_get', {'group_uid': '3d0b4432-7390-4ae5-92be-52c3e39dc775'}, 'WatcherSchemaResponse', 'https://api.intel471.com/v1/watcherGroups/3d0b4432-7390-4ae5-92be-52c3e39dc775/watchers'),
+    'WatchersApi:watcher_groups_group_uid_watchers_watcher_uid_get': ('WatchersApi', 'watcher_groups_group_uid_watchers_watcher_uid_get', {'group_uid': '3d0b4432-7390-4ae5-92be-52c3e39dc775', 'watcher_uid': '81808755feef2000acb593a3a5239cdc'}, 'WatcherSchema', 'https://api.intel471.com/v1/watcherGroups/3d0b4432-7390-4ae5-92be-52c3e39dc775/watchers/81808755feef2000acb593a3a5239cdc'),
+    'YARAApi:yara_get': ('YARAApi', 'yara_get', {'threat_type': 'malware', 'count': 1}, 'YARASearchResponse', 'https://api.intel471.com/v1/yara'),
+}
+
+@patch('titan_client.rest.RESTClientObject')
+@pytest.mark.parametrize('api_cls_name, method_name, kwargs, filename, query_url', test_params.values(), ids=test_params.keys())
+def test_api_responses(rest_client_class_mock, api_cls_name, method_name, kwargs, filename, query_url):
+    rest_client_response = MagicMock()
+    rest_client_response.status = 200
+    rest_client_response.reason = 'OK'
+    rest_client_response.getheader.return_value = 'application/json; charset=utf-8'
+    response = read_fixture(f'{PREFIX}/fixtures/api_responses/{filename}.json')
+    rest_client_response.data = json.dumps(response).encode('utf-8')
+
+    rest_client_instance_mock = MagicMock()
+    rest_client_instance_mock.GET.return_value = rest_client_response
+
+    rest_client_class_mock.side_effect = [rest_client_instance_mock]
+
+    with titan_client.ApiClient(configuration) as api_client:
+        api_instance = getattr(titan_client, api_cls_name)(api_client)
+        api_response = getattr(api_instance, method_name)(**kwargs)
+        assert rest_client_instance_mock.GET.call_args_list[0][0][0] == query_url
+        if 'uid' in method_name:
+            assert rest_client_instance_mock.GET.call_args_list[0][1]['query_params'] == []
+        else:
+            assert rest_client_instance_mock.GET.call_args_list[0][1]['query_params'] == [(re.sub('_.',lambda x: x.group()[1].upper(), k), v) for k,v in kwargs.items()]
```

Comment:
Had to use `call_args_list` instead of `mock_calls` to support backward compatibility for Python 3.6
Python 3.6 and 3.7 does not support the following original code
```
assert rest_client_instance_mock.GET.mock_calls[0].args[0] == query_url
if 'uid' in method_name:
    assert rest_client_instance_mock.GET.mock_calls[0].kwargs['query_params'] == []
else:
    assert rest_client_instance_mock.GET.mock_calls[0].kwargs['query_params'] == [(re.sub('_.',lambda x: x.group()[1].upper(), k), v) for k,v in kwargs.items()]
```

Reply:
Nice. I like the approach to transforming snake_case to CamelCase. Actually I didn't know that `re.sub` accepts a callable as a second arg.

I would refactor it a bit to make it more readable:

```
if 'uid' in method_name:
    expected_params = []
else:
    kwargs_camel_case = [(re.sub('_.', lambda x: x.group()[1].upper(), k), v) for k, v in kwargs.items()]
    expected_params = kwargs_camel_case
assert rest_client_instance_mock.GET.call_args_list[0][0][0] == query_url
assert rest_client_instance_mock.GET.call_args_list[0][1]['query_params'] == expected_params
```

This expression `[(re.sub('_.', lambda x: x.group()[1].upper(), k), v) for k, v in kwargs.items()]` is a bit cryptic at a first glance and it would be worth to store its results in a variable that explains what's going on (like `kwargs_camel_case`). This way it's easy to follow the flow when reading the function, without going into details.